pkg/process/metadata/workloadmeta/extractor.go

```diff
@@ -66,6 +71,14 @@ var (
 		subsystem, "diffs_dropped", "The number of diffs dropped due to channel contention")
 )
 
+// GetSharedWorkloadMetaExtractor returns a shared WorkloadMetaExtractor
+func GetSharedWorkloadMetaExtractor(sysprobeConfig config.Reader) *WorkloadMetaExtractor {
+	initWorkloadMetaExtractor.Do(func() {
+		sharedWorkloadMetaExtractor = NewWorkloadMetaExtractor(sysprobeConfig)
+	})
+	return sharedWorkloadMetaExtractor
+}
```

Comment:
Could you provide a bit more context on why we need this new logic? It seems that the only one calling this function is coming from `comp/core/workloadmeta/collectors/internal/process/process_collector.go`

Mostly I want to understand if it justified the use of `sync.Once` or not.

Reply:
This was created so that the process check and process collector can share language detection data when they are both running via `processDiffCh`. I removed the call in the process check for now in this PR as the collector is meant to be unused until https://github.com/DataDog/datadog-agent/pull/28172 is merged. [Here's](https://github.com/DataDog/datadog-agent/pull/27559/commits/8c34e915fd1a180b443bfab481fd36a9a5161325) how it will be used.

Reply:
Thank for the clarification @daniel-taf